src/controller/ReadInteraction.h

```diff
@@ -177,13 +188,14 @@ template <typename DecodableEventType>
 CHIP_ERROR ReportEvent(Messaging::ExchangeManager * apExchangeMgr, EndpointId endpointId,
                        ReportEventParams<DecodableEventType> && readParams)
 {
-    ClusterId clusterId = DecodableEventType::GetClusterId();
-    EventId eventId     = DecodableEventType::GetEventId();
-    app::EventPathParams eventPath(endpointId, clusterId, eventId);
+    ClusterId clusterId                  = DecodableEventType::GetClusterId();
+    EventId eventId                      = DecodableEventType::GetEventId();
     app::InteractionModelEngine * engine = app::InteractionModelEngine::GetInstance();
     CHIP_ERROR err                       = CHIP_NO_ERROR;
 
-    readParams.mpEventPathParamsList    = &eventPath;
+    std::unique_ptr<app::EventPathParams[]> readPaths(Platform::New<app::EventPathParams>());
```

Comment:
Similar issues here as for the attribute case.

Reply:
Updated, thanks